skyplane/obj_store/hdfs_interface.py

```diff
@@ -0,0 +1,49 @@
+import os
+from pyarrow import fs
+from dataclasses import dataclass
+from skyplane.obj_store.file_system_interface import FileSystemInterface, LocalFile
+from typing import Iterator, List, Optional, Tuple
+
+
+@dataclass
+class HDFSFile(LocalFile):
+    path: str
+    size: Optional[int] = None
+    last_modified: Optional[str] = None
+    file_format: Optional[str] = None
+
+
+class HDFSInterface(FileSystemInterface):
+    def __init__(self, path, port=8020):
+        self.path = path
+        self.port = port
+        self.hdfs = fs.HadoopFileSystem(host=self.path, port=self.port, extra_conf={"dfs.permissions.enabled": "false"})
+
+    def path(self) -> str:
```

Comment:
This can exist and I see you don't use it other than to return the path, but ideally we should get rid of this. 
HDFS interface should be stateless with respect to files, but can be stateful with respect to the hdfs connection.

Reply:
The path function returns he master public DNS address, so it returns the path of the HDFS connection. Is that also not advisable?